src/utils/getUrlByRequestOptions.test.ts

```diff
@@ -14,6 +14,20 @@ test('returns a URL based on the basic RequestOptions', () => {
   expect(url).toHaveProperty('href', 'https://127.0.0.1/resource')
 })
 
+test('resolves protocol from agent, if one exists', () => {
+  const options: RequestOptions = {
+    host: '127.0.0.1',
+    path: '/',
+    agent: new Agent()
+  }
+  const url = getUrlByRequestOptions(options)
+
+  expect(url).toBeInstanceOf(URL)
+  expect(url).toHaveProperty('protocol', 'https:')
```

Comment:
This used to be `http:` in the past, correct?

Since it doesn't have an explicit `protocol` set, we defaulted to `http:`, while this particular case used `https.Agent`.